public/index.html

```diff
@@ -0,0 +1,187 @@
+                >/api/shows?country=Sweden</a
+              >
+            </p>
+          </li>
+          <li>
+            <p class="endpoint-description">
+              Get all shows with a specific rating
+            </p>
+            <p>
+              e.g.<a class="endpoint-url" href="/api/shows?rating=TV-14"
+                >/api/shows?rating=TV-14</a
+              >
+            </p>
+          </li>
+          <li>
+            <p class="endpoint-description">
+              Get all shows with a specific category
+            </p>
+            <p>
+              e.g.<a class="endpoint-url" href="/api/shows?category=Comedies"
+                >/api/shows?category=Comedies</a
+              >
+            </p>
+          </li>
+          <li>
+            <p class="endpoint-description">
+              Get all shows with a specific description
+            </p>
+            <p>
+              e.g.<a
+                class="endpoint-url"
+                href="/api/shows?description=In this sequel to the 2016"
+                >/api/shows?description=In this sequel to the 2016</a
+              >
+            </p>
+          </li>
+          <li>
+            <p class="endpoint-description">
+              Get all shows with a specific show type
+            </p>
+            <p>
+              e.g.<a class="endpoint-url" href="/api/shows?type=movie"
+                >/api/shows?type=movie</a
+              >
+            </p>
+          </li>
+          <li>
+            <p class="endpoint-description">Get all movies with pagination</p>
+            <p>
+              e.g.<a class="endpoint-url" href="/api/shows?page=1"
+                >/api/shows?page=1</a
+              >
+            </p>
+          </li>
+          <li>
+            <p class="endpoint-description">
+              Get shows with several parameters
+            </p>
+            <p>
+              e.g.<a class="endpoint-url" href="/api/shows?actor=Anna&year=2019"
+                >/api/shows?actor=Anna&year=2019</a
+              >
+            </p>
+          </li>
+          <li>
+            <p class="endpoint-description">
+              Get shows with several parameters
+            </p>
+            <p>
+              e.g.<a
+                class="endpoint-url"
+                href="/api/shows?actor=Anna&year=2019&duration=24"
+                >/api/shows?actor=Anna&year=2019&duration=24</a
+              >
+            </p>
+          </li>
+          <li>
+            <p class="endpoint-description">
+              Get shows with several parameters (error example)
+            </p>
+            <p>
+              e.g.<a
+                class="endpoint-url"
+                href="/api/shows?actor=Anna&duration=105&page=3"
+                >/api/shows?actor=Anna&duration=105&page=3</a
+              >
+            </p>
+          </li>
+        </ul>
+      </section>
+    </main>
+  </body>
+</html>
```

Comment:
😍 😍 😍 😍 😍 😍 😍 DOCUMENTATION 😍 😍 😍 😍 😍 😍 😍 

Is there any way to auto-generate it... 🤔 ?

Reply:
I know of at least three different tools/frameworks that can make your life a bit easier when it comes to API documentation:

- swagger.io
- raml.org
- mulesoft.com

Maybe I will have time to implement one of these technologies in my final project.